### ISSUE TITLE
low performance for full supervised setting

### ISSUE THREAD

**Issue by zhangyuygss** [yassouali/CCT#1]
I modified the config file to set the code to 'supervised' mode, but the result seems to be very low: 
`
Epoch : 40 | Mean_IoU : 0.699999988079071 | PixelAcc : 0.933 | Val Loss : 0.26163
`
compared with 'semi' mode:

`
Epoch : 40 | Mean_IoU : 0.7120000123977661 | PixelAcc : 0.931 | Val Loss : 0.31637
`

Note that I have changed the supervised list to the 10k+ augmented list in the 'supervised' setting.
Did I miss something here?

**Comment by yassouali** [comment]
Hi, thank you for your interest in our work,

To make sure I understand, you are training on all of the labels in a supervised manner, and the obtained mIoU is low at 40 epochs ?

Ill run the same on my end and I will get back to you.

**Comment by zhangyuygss** [comment]
Yes, that's my setting. What's your supervised result with all the labeled data?

**Comment by yassouali** [comment]
Thanks,

If I remeber corretly, it was in the high seventies, but it was an earlier version of the code used for the paper, I didnt test it with this version, let me get back to you on this.

**Comment by yassouali** [comment]
Hi, I've just retrained for 35 out of 100 and gotten 73 miou, I expect to go a little bit higher before the end of training, 

```
         val_loss       : 0.21543
         Pixel_Accuracy : 0.937
         Mean_IoU       : 0.7310000061988831
```

I did actually find an error in the code that might lead to a lower number of epochs in your case, for the supervised mode, the number of iterations is calculated in terms of the unsupervised examples:
```
tbar = tqdm(range(len(self.supervised_loader)), ncols=135)
```
should be:
```
tbar = tqdm(range(len(self.supervised_loader)), ncols=135)
```

so in your case, the number of the unsupervised examples is lower (depending on number of examples you're training on), so this reduced the number of epochs, so for example, the result you're obtained for 40 epochs is, in reality, say only 20 epochs going through the whole labeled data, so you can simply train for longer and you'll get better results. I just pushed the correction for this bug, and if you run training for 100 epochs, I expect you'll get 74 / 75 at the end. And you can always gain a bit more in mIoU with semi mode or weakly mode.

I hope I answered your question.
Thanks.

**Comment by zhangyuygss** [comment]
Thanks,
I suppose you mean 
`tbar = tqdm(range(len(self.unsupervised_loader)), ncols=135)`
to
`tbar = tqdm(range(len(self.supervised_loader)), ncols=135)`
here. 
I re-run the code and also get 73.4 mIoU at epoch 60.

**Comment by yassouali** [comment]
Yes exactly, thank you !
